### PR TITLE
add 4 pin crystal symbol

### LINKED ISSUE #202: Add four pin crystal variant
Currently we only have a two-pin crystal

Image

/bounty $5

### PR DESCRIPTION
**Pull request by ni9999**
fixes #202 
/claim #202

### CODE REVIEW

**Review by seveibar — CHANGES_REQUESTED** [review]
nice work, anchors aren't quite right in either view. Make sure the text grows away from the symbol. These texts are replaced with potentially very long names so they must be anchored properly

Use `modifySymbol` it's a better, more modern api

image

**Comment by seveibar** [comment]
@ni9999 nice work, was happy to see this

File: symbols/crystal_4pin_horz.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  ] as Primitive[],
+  ports: [
+    { ...refblocks.top1, labels: ["2"] }, // TODO add more "standard" labels
+    { ...refblocks.bottom1, labels: ["4"] }, // TODO add more "standard" labels
+    { ...refblocks.left1, labels: ["1"] }, // TODO add more "standard" labels
+    { ...refblocks.right1, labels: ["3"] }, // TODO add more "standard" labels
+  ],
+  size: { width: bounds.width, height: bounds.height },
+  center: { x: bounds.centerX, y: bounds.centerY },
+})
```
[the remark this answers is not in the document]
Comment (ni9999) [reply symbols/crystal_4pin_horz.ts]: @seveibar changed accordingly.

File: symbols/crystal_4pin_horz.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import svgJson from "assets/generated/crystal_4pin.json"
+import { modifySymbol } from "drawing/modify-symbol/modify-symbol"
+
+export default modifySymbol(svgJson)
+  .labelPort("top1", ["gnd1"])
+  .labelPort("bottom1", ["gnd2"])
```
Comment (seveibar) [inline symbols/crystal_4pin_horz.ts:6]:
```suggestion
  .labelPort("top1", ["2", "gnd1"])
  .labelPort("bottom1", ["4", "gnd2"])
```

**Review by seveibar — CHANGES_REQUESTED** [review]
@ni9999, just accept my suggestion and regenerate the snapshot, and you're good to go!!!

**Comment by ni9999** [comment]
@seveibar accepted but now gnd1 and gnd2 is gone
crystal_4pin_horz

**Review by seveibar — APPROVED** [review]
Yes theyre now alias rather than primary labels, we should show all the labels but it would be crowded 🤷

**Comment by seveibar** [comment]
@ni9999 im eager to merge this but it has conflicts!

**Pull request merged**